codalab/rest/worksheets.py

```diff
@@ -211,24 +212,29 @@ def create_worksheet_items():
     |replace| - Replace existing items in host worksheets. Default is False.
     """
     replace = query_get_bool('replace', False)
-
     new_items = WorksheetItemSchema(strict=True, many=True).load(request.json).data
 
-    worksheet_to_items = {}
-    for item in new_items:
-        worksheet_to_items.setdefault(item['worksheet_uuid'], []).append(item)
-
-    for worksheet_uuid, items in worksheet_to_items.items():
-        worksheet_info = get_worksheet_info(worksheet_uuid, fetch_items=True)
+    if len(new_items) > 0:
```

Comment:
I'd try to avoid the branching, because there's nothing special about `new_items` and more code paths makes it harder to maintain.

The reason this is more complicated than it needs to be is because in the current code, each item in `new_items` is associated with the worksheet that it should be added to, rather than having a single `uuid` where all of `new_items` gets added to.  Can we change all the places where `client.create('worksheet-items', ...` is called to obey the logic where we pass in a uuid and a list of items (which themselves don't have the associated uuid).  This should vastly simply things.  (I don't know why we had this more complicated and powerful way of specifying worksheet uuids...)

Reply:
Yes, I agree. Always passing the `uuid` in would be a cleaner way to handle detached items. I think I should also check if there is any other function is currently using `create_worksheet_items` and see how they will be affected by this change.